### PR TITLE
authentication - facebook and traditional login and signup and account

### PR DESCRIPTION
**Pull request by peta15**
management

### CODE REVIEW

File: app/components/control-messages.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  inputs: ['controlName: control'],
+  template: `<div *ngIf="errorMessage !== null">{{errorMessage}}</div>`
+})
+export class ControlMessages {
+  controlName: string;
+  constructor( @Host() private _formDir: NgFormModel) { }
+
+  get errorMessage() {
+    // Find the control in the Host (Parent) form
+    let c = this._formDir.form.find(this.controlName);
```
Comment (rivertam) [inline app/components/control-messages.ts]: This should probably be `const`.

File: app/pages/devices/edit_device.ts [hunk 2]
```diff
@@ -37,7 +37,10 @@ export class EditDevicePage {
   }
 
   save(): void {
-    this.deviceService.edit(this.device.objectId, this.changedValues).then(res => this.nav.pop());
+    var changedValues = this.changedValues
```
Comment (rivertam) [inline app/pages/devices/edit_device.ts]: same

File: app/components/control-messages.ts (hunk 1 above)
Comment (vinpez) [inline app/components/control-messages.ts]: Ben, stop trolling my repo unless you want to start working on it.

Comment (rivertam) [inline app/components/control-messages.ts]: I'm helping

Comment (rivertam) [inline app/components/control-messages.ts]: :^)

**Pull request merged**